src/promptflow/promptflow/_core/openai_injector.py

```diff
@@ -133,11 +214,27 @@ def inject_openai_api():
     It stores the original methods as _original attributes of the create methods.
     2. Updates the openai api configs from environment variables.
     """
-    for openai_api in available_openai_apis():
+    for openai_api_sync in available_openai_apis_sync():
         # Check if the create method of the openai_api class has already been modified
-        if not hasattr(openai_api.create, "_original"):
+        if hasattr(openai_api_sync, "create") and not hasattr(openai_api_sync.create, "_original"):
             # If not, modify it by calling the inject function with it as an argument
-            openai_api.create = inject(openai_api.create)
+            openai_api_sync.create = inject_sync(openai_api_sync.create)
+
+    for openai_api_async in available_openai_apis_async():
+        # Due to the issue reported in https://github.com/openai/openai-python/issues/996,
+        # Async interface was not recognized as a coroutine by inspect.iscoroutinefunction.
+        # This means that async injection needs to be handled differently to ensure decorators
+        # behave correctly for asynchronous methods.
+
+        # Check if the create method of the openai_api class has already been modified
+        if hasattr(openai_api_async, "acreate") and not hasattr(openai_api_async.acreate, "_original"):
+            # If not, modify it by calling the inject function with it as an argument
+            openai_api_async.acreate = inject_async(openai_api_async.acreate)
```

Comment:
Add `if IS_LEGACY_OPENAI` to this block?

Reply:
OK. I will add the block